### PR TITLE
Get rid of blueprint miner entity hack

### PR DESCRIPTION
**Pull request by slipher**
A temporary leech entity will no longer be constructed when calculating the potential efficiency of a new miner.

Also implement the TODO to hide under-construction enemy miners from potential efficiency calculations.

### CODE REVIEW

File: src/sgame/components/MiningComponent.h [hunk 1]
```diff
@@ -47,15 +46,16 @@ class MiningComponent: public MiningComponentBase {
 		 */
 		static float InterferenceMod(float distance);
 
-		/**
```
Comment (DolceTriade) [inline src/sgame/components/MiningComponent.h:50]: Why was this remove?

File: src/sgame/components/MiningComponent.h [hunk 2]
```diff
@@ -47,15 +46,16 @@ class MiningComponent: public MiningComponentBase {
...
-		 * @return Whether the miner is currently mining (and interfering with other miners).
-		 */
-		bool Active() { return active; }
+		struct Efficiencies {
```
Comment (DolceTriade) [inline src/sgame/components/MiningComponent.h:49]: Given that their usage is mutually exclusive, would it be better to expose two functions to fetch these values?

**Review by DolceTriade — COMMENTED** [review]
nice fix for a hack

File: src/sgame/components/MiningComponent.h (hunk 1 above)
Comment (slipher) [reply src/sgame/components/MiningComponent.h:50]: Ended up being unused

File: src/sgame/components/MiningComponent.h (hunk 2 above)
Comment (slipher) [reply src/sgame/components/MiningComponent.h:49]: We have these cases:
- Blueprint miner: calculate only predicted efficiency
- Miner is constructed: need to update only predicted efficiencies
- Miner finishes construction: need to update only current efficiencies
- Miner dies: need to update both predicted and current efficiencies

I could have flags for which ones InformNeighbors should update, but that seemed overly complex so I stuck with always updating both, which is done together for efficiency.

Comment (DolceTriade) [reply src/sgame/components/MiningComponent.h:49]: Got it.

**Review by DolceTriade — APPROVED** [review]

**Pull request merged**